lib/amazon_mws.rb

```diff
@@ -66,15 +69,16 @@ def confirm_order
     })
   end
 
-  def authorize(ref_number, total, currency)
-    process({
-      "Action"=>"Authorize",
-      "AmazonOrderReferenceId" => @number,
-      "AuthorizationReferenceId" => ref_number,
-      "AuthorizationAmount.Amount" => total,
-      "AuthorizationAmount.CurrencyCode" => currency,
-      "TransactionTimeout" => 0
-    })
+  def authorize(ref_number, total, currency, seller_authorization_note: nil)
+    client.authorize(
+      @number,
```

Comment:
> Oh it's the order number

Well, it's the Amazon order_reference_id, is that what you mean?  (Not the `spree_orders.number`).  I've gone ahead and renamed `@number` to `amazon_order_reference_id` and `ref_number` to `authorization_reference_id`, just like they do [in the amazon gem itself](https://github.com/amzn/login-and-pay-with-amazon-sdk-ruby/blob/5202d4fc42f0968c7835a28214f515bb566e836c/lib/pay_with_amazon/client.rb#L494-L496).

> There's a lot of refactoring that we need in general in all the extension

Very 👍 on that.  The whole concept of the AmazonMws object seems weird to me.  My thought was that we first work on getting all the plumbing updated inside `AmazonMws` to use the Amazon gem, and then once that's done just take a pass and try to get rid of the `AmazonMws` object entirely (or significantly clean it up).